### PR TITLE
Remove AZ logic

### PR DESCRIPTION
**Pull request by ijsnellf**
Removes the majority of the AZ logic. 

Removing as part of: https://github.com/istio/istio/issues/6049. Once this is merged, it should be easy to nuke the v1 discovery server.

### CODE REVIEW

File: pilot/pkg/model/service.go [hunk 1]
```diff
@@ -354,6 +354,7 @@ const (
 // - consul: defaults to 'instance.Datacenter'
 //
 // This is used by EDS to group the endpoints by AZ and by .
+// TODO: remove me?
```
Comment (ijsnellf) [inline pilot/pkg/model/service.go]: this is used by EDS here: https://github.com/istio/istio/blob/release-1.0/pilot/pkg/proxy/envoy/v2/eds.go#L216

Comment (ijsnellf) [reply pilot/pkg/model/service.go]: not sure if we want to keep the EDS logic for something in the future

File: pilot/cmd/pilot-agent/main.go [hunk 2]
```diff
@@ -50,7 +50,7 @@ var (
 	configPath               string
 	binaryPath               string
 	serviceCluster           string
-	availabilityZone         string
+	availabilityZone         string // TODO: remove me?
```
[the remark this answers is not in the document]
Comment (rshriram) [reply pilot/cmd/pilot-agent/main.go]: No. Let’s keep it.

File: pilot/pkg/model/service.go (hunk 1 above)
Comment (rshriram) [reply pilot/pkg/model/service.go]: Also keep.

**Pull request merged**